lib/ansible/cli/vault.py

```diff
@@ -181,7 +181,7 @@ def run(self):
             if not vault_secrets:
                 raise AnsibleOptionsError("A vault password is required to use Ansible's Vault")
 
-        if self.action in ['encrypt', 'encrypt_string', 'create', 'edit']:
+        if self.action in ['encrypt', 'encrypt_string', 'create']:
```

Comment:
So following on from my comment in #35834 I thought you added the edit action in PR #31067 to solve #30491

I'm not saying either way is right or wrong, simply asking the question seeing as an issue of the same type introduced a fix, now that code is being removed. Just trying to provide context here.

Reply:
In #31067 I thought I was going to try to let the encrypt-vault-id be chosen/overridden for 'edit' but have since decided not to allow that and to always use the decrypt vault-id as the encrypt vault-id for 'edit'.